src/uint256.h

```diff
@@ -130,7 +130,7 @@ class uint160 : public base_blob<160> {
  */
 class uint256 : public base_blob<256> {
 public:
-    uint256() {}
+    uint256() : base_blob<256>() {}
```

Comment:
Why it's here? The following simple example:
```
#include <iostream>
#include <iomanip>
#include <cstring>

/** Template base class for fixed-sized opaque blobs. */
template<unsigned int BITS>
class base_blob
{
protected:
    // static constexpr int WIDTH = BITS / 8;
    // uint8_t m_data[WIDTH];
    enum { WIDTH=BITS/8 };
    alignas(uint32_t) uint8_t m_data[WIDTH];
public:
    //constexpr base_blob() : m_data() {}
    base_blob()
    {
        std::cout << __FILE__ << "," << __LINE__ << ": " << __func__ << std::endl;
        memset(m_data, 0, sizeof(m_data));
    }
};

class uint256 : public base_blob<256> {
public:
    //constexpr uint256() {}
    uint256() {
        std::cout << __FILE__ << "," << __LINE__ << ": " << __func__ << std::endl;
    }
    void PrintData()
    {
        for (int i = 0; i < WIDTH; ++i) std::cout << std::setfill('0') << std::setw(2) << std::hex << int(m_data[i]) << " ";
        std::cout << std::endl;
    }
    bool IsNull() const
    {
        for (int i = 0; i < WIDTH; i++)
            if (m_data[i] != 0)
                return false;
        return true;
    }
};

int main() {
    uint256 a;
    a.PrintData();

    return 0;
}
```
Shows us that the ctor of `base_blob` is called anyway even without call base class ctor in initializer list, or i miss something? The result of the sample above will be:
```
/home/decker/nullptr.cpp,19: base_blob
/home/decker/nullptr.cpp,28: uint256
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
```

Reply:
I added that because I thought the base constructor was not being called and garbage was in the array. It turns out the garbage was not caused by what I thought it was. I will back this out.